scripts/publish.js

```diff
@@ -49,13 +49,28 @@ async function publish(pkgs, dryRun = true) {
   })
 }
 
+/**
+ * A JSON.parse that doesn't mind a rejected promise
+ * @param {Promise<string>} promise
+ * @returns {Promise<any>}
+ */
+const chillJSONParse = async (promise) =>
+  promise
+    .catch((err) => {
+      console.error(err.message)
+      return '{}'
+    })
+    .then(JSON.parse)
+
```

Comment:
@naugtur So if this parsing fails, either a) `package.json` is invalid, or b) `npm show --json` is invalid, and we should continue _but_ with a message to the user that they may need to publish this package manually. 

I don't know why we'd want to continue normally if a `package.json` is invalid though.

Reply:
If package.json is not found or invalid, the result of parsing is an empty object, the error was printed and the package was skipped.
I introduced it meaning to complete publishing what makes sense and let you fix and rerun to publish the thing that was broken before. 
But I think I was making a wrong assumption - if we throw an error from parsing package.json - the whole publish is not happening and the release can be fixed before another attempt.  
I've lost confidence in my alleged improvement :sweat_smile:

Reply:
Let's drop the chillJSONParse function and use regular JSON.parse in a big try-catch in the map callback to add a message explaining the package from the folder was not fit to publish and then adding the rest of the error for debugging.